apps/tangle-dapp/containers/TransferTxContainer/TransferTxContainer.tsx

```diff
@@ -122,28 +143,52 @@ const TransferTxContainer: FC<TransferTxContainerProps> = ({
               {txError.message}
             </Typography>
           )}
+
+          {!isValidReceiverAddress && receiverAddress !== '' && (
+            <Typography variant="body1" fw="normal" className="!text-red-50">
+              Invalid receiver address
+            </Typography>
+          )}
         </div>
 
-        <ModalFooter className="flex flex-col gap-1 px-8 py-6">
-          <Button
-            isFullWidth
-            isDisabled={!canInitiateTx || executeTransferTx === null}
-            isLoading={!isReady}
-            loadingText={getTxStatusText(status)}
-            onClick={executeTransferTx !== null ? executeTransferTx : undefined}
-          >
-            Send
-          </Button>
-
-          <Link href={TANGLE_DOCS_URL} target="_blank">
-            <Button isFullWidth variant="secondary">
-              Learn More
+        <ModalFooter className="flex items-center gap-2 px-8 py-6 space-y-0">
+          <div className="flex-1">
+            <Button
+              isFullWidth
+              isDisabled={
+                !canInitiateTx ||
+                executeTransferTx === null ||
+                !isValidReceiverAddress
+              }
+              isLoading={!isReady}
+              loadingText={getTxStatusText(status)}
+              onClick={
+                executeTransferTx !== null ? executeTransferTx : undefined
+              }
+            >
+              Send
             </Button>
-          </Link>
+          </div>
+
+          <div className="flex-1">
+            <Link href={TANGLE_DOCS_URL} target="_blank" className="w-full">
+              <Button isFullWidth variant="secondary">
+                Learn More
+              </Button>
+            </Link>
+          </div>
         </ModalFooter>
       </ModalContent>
     </Modal>
   );
 };
 
 export default TransferTxContainer;
+
+/** @internal */
+function getTypedChainIdFromAddr(addr?: string | null): number {
+  if (!addr) return NaN;
```

Comment:
If the address is not ready yet, consider returning null. This forces the caller to handle the case in which the address hasn't loaded, and it is clear on the type system.